eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -233,26 +234,45 @@ isValidAlonzoTxL = lens isValid (\tx valid -> tx {isValid = valid})
 {-# INLINEABLE isValidAlonzoTxL #-}
 
 deriving instance
-  (Era era, Eq (Core.TxBody era), Eq (TxWits era), Eq (AuxiliaryData era)) =>
+  ( Era era,
+    Eq (Core.TxBody era),
+    Eq (TxAuxData era),
+    Eq (TxWits era)
+  ) =>
```

Comment:
There is no need to make this multiline
```suggestion
  (Era era, Eq (Core.TxBody era), Eq (TxWits era), Eq (TxAuxData era)) =>
```

Reply:
I never understood whether ormulu has opinions on this -  and this comment prompted me to try to understand.
Looks like if you split it on multilines, ormulu respects that . But on the other hand, it doesn't seem to know to split the line itself beyond a certain number of characters, which is what I  would have expected! 

So for example this has been "ormollized" successfully: 

![2022-09-12_19-07](https://user-images.githubusercontent.com/2070003/189725516-3ec32973-6ee6-4f1d-ad11-3154e68a5963.png)

So I wonder then what is our convention for line width?  Is there an implicit one we're having?

Reply:
That's a good question. However, ormolu is only tangentially related to this question, because it formats the code, but it does not enforce the line length, since that is an impossible problem, because it would require customization, which ormolu does not support. I'll give you my personal guideline and personal motivation behind it:

* Using 80 character limit was imposed in the early days when terminals ability was limited to displaying only 80 characters in one line, so this archaic guideline is no longer relevant
* What is relevant, however, is that many people, including me, use split screen. Eg. two emacs buffers side-by-side, or an editor and a terminal. And using 80 characters works pretty good. I also found that using something like 90 to a 100 character lines work even better and causes no wraparound while using all default OS settings, eg. font size etc. So I'd say around 90 character limit is really a sweet spot.
* Using lines much longer than 100 characters, like in the example above, not only makes it impossible to read in some editors, but also it makes it impossible to review a PR with such a change, because Github does not wraparound when showing the diff, so it will just cut off right half of that diff
* Using lines that are too short on the other hand where the whole thing could fit into one slightly longer line utilizes the space least efficiently and makes you scroll through many more lines in order to be able to read the code. And as I am sure we all know, more lines of code does not mean better code :smile: 

All of those point are not a hard guideline, of course, code readability and conventions must be the priority. 

For instance, using semicolons is absolutely legal in Haskell, however it is unconventional, thus it is extremely rare when you'll see someone using them, consequently making code that uses them less readable. I can best demonstrate my point on an example, perhaps one that I was currently working on. Which one do you think is more readable?

1. The original:

```haskell
instance (FromCBOR a, FromCBOR b, FromCBOR c) => FromCBOR (a, b, c) where
  fromCBOR =
    decodeListLenOf 3
    !x <- fromCBOR
    !y <- fromCBOR
    !z <- fromCBOR
    return (x, y, z)
```

2. or this obscure variant?
```haskell
instance (FromCBOR a, FromCBOR b, FromCBOR c) => FromCBOR (a, b, c) where
  fromCBOR = do decodeListLenOf 3; !x <- fromCBOR; !y <- fromCBOR; !z <- fromCBOR; return (x, y, z)
```
3. or even better
```haskell
instance (FromCBOR a, FromCBOR b, FromCBOR c) => FromCBOR (a, b, c) where fromCBOR = do decodeListLenOf 3; !x <- fromCBOR; !y <- fromCBOR; !z <- fromCBOR; return (x, y, z)
```

4. Another extreme, which is suggested in this PR, is to use more lines. Problem with this approach is that it does not utilize available screen space efficiently, thus making the whole module less readable, especially when it is all written in such style:
```haskell
instance
  ( FromCBOR a,
    FromCBOR b,
    FromCBOR c
  ) =>
  FromCBOR (a, b, c)
  where
  fromCBOR = do
    decodeListLenOf 3
    !x <- fromCBOR
    !y <- fromCBOR
    !z <- fromCBOR
    return (x, y, z)
```

5. In my opinion the optimal would be to use `Applicative` syntax, thus making it less verbose, more readable and using less space:

```haskell
instance (FromCBOR a, FromCBOR b, FromCBOR c) => FromCBOR (a, b, c) where
  fromCBOR = do
    decodeListLenOf 3
    (,,) <$> fromCBOR <*> fromCBOR <*> fromCBOR
```

6. Below is the best variant, in my opinion, which is quite readable for anyone who is used to Applicative style parsers, but that is obviously quite subjective:
```haskell
instance (FromVCBOR a, FromVCBOR b, FromVCBOR c) => FromVCBOR (a, b, c) where
  fromVCBOR = decodeListLenOf 3 *> ((,,) <$> fromVCBOR <*> fromVCBOR <*> fromVCBOR)
```

Although, IMHO, the original is good enough.

Point I am trying to make is that I was able to put on two lines and keep it readable the same functionality, which would go onto 13 lines if it was formatted in a multi-line hungry fashion.

So the guideline is: the more code you can see at once while keeping it readable for most readers and following the Haskell conventions and not upsetting ormolu, the better.

@JaredCorduan if you agree with above, should I put this into `CONTRIBUTING.md`  doc or start a new code styling guideline doc? I can certainly place more similar stuff in such document over time.

Reply:
I agree :100: with you, @lehins ,  on all of the above. The only thing I would add is: when in doubt, favor vertical space since it plays better with source control. I think this is great material for the `CONTRIBUTING.md` doc.

@teodanciu - ormolu gives the programmer the option between "all on one line" or "on separate lines in the style ormolu chooses".

Reply:
I m pretty sure this was done by ormolou. But I changed it back. We will see what happens.